core/arch/arm/kernel/boot.c

```diff
@@ -914,58 +914,82 @@ static uint64_t get_dt_val_and_advance(const void *data, size_t *offs,
 	return rv;
 }
 
-static struct core_mmu_phys_mem *get_memory(void *fdt, size_t *nelems)
+static int get_memory_all(void *fdt, struct core_mmu_phys_mem *mem)
 {
-	int offs = 0;
-	int addr_size = 0;
-	int len_size = 0;
-	size_t prop_len = 0;
 	const uint8_t *prop = NULL;
+	unsigned long long a = 0;
+	unsigned long long l = 0;
```

Comment:
`uint64_t` would be better here, since it is the return type of `get_dt_val_and_advance()`